### ISSUE TITLE
Add CI

### ISSUE THREAD

**Issue by kupiakos** [danielhenrymantilla/rust-uninit#33]
Ensure that at the very least:

- `cargo test` works
- `cargo +nightly test --all-features` works

It could also:
- Check the MSRV
- Check formatting
- Check clippy

**Comment by danielhenrymantilla** [comment]
Yes, this is so key; I guess back in the day I got lazy and was just `cargo test`ing myself 😱.

This should be top-priority before anything else (at least the `test`ing part; the extra features such as checking formatting would have to be embedded within their own "format properly the codebase" efforts
